pkg/cmd/cli/config/store.go

```diff
@@ -0,0 +1,185 @@
+package config
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/golang/glog"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	clientcmdapi "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/api"
+
+	"github.com/openshift/origin/pkg/cmd/flagtypes"
+)
+
+const (
+	fromKube      = "fromkube"
+	fromOpenShift = "fromopenshift"
+)
+
+// A ConfigStore is the representation of a config from one individual config file. Can be used
+// to persist configs by being explicit about the file to save.
+type ConfigStore struct {
+	Config         *clientcmdapi.Config
+	Path           string
+	providerEngine string
+}
+
+type ConfigStoreNotFoundError struct {
+	msg string
+}
+
+func (e *ConfigStoreNotFoundError) Error() string {
+	return e.msg
+}
+
+func (c *ConfigStore) FromOpenShift() bool {
+	return c.providerEngine == fromOpenShift
+}
+
+func (c *ConfigStore) FromKube() bool {
+	return c.providerEngine == fromKube
+}
+
+// Load a ConfigStore, first trying the explicit path to a config file provided as argument,
+// and if it was not provided, load using the priority conventions used by the
+// ClientConfigLoadingRules. Error if none can be found.
+func Load(clientCfg *client.Config, explicitPath string) (*ConfigStore, error) {
+	// if an explicit path were provided only tries it
+	path := explicitPath
+	if len(path) > 0 {
```

Comment:
Make a separate method for this case.  The arguments are different and the intent is different.  In one case, the caller wants exactly one file or nothing.  In the other case, he wants "load me something from whatever default you think is appropriate".